mitodl/mit-open
# Updates to ChoiceBox; Checkbox, Radio components

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4661

### Description (What does it do?)

Updates the styles and hover states for the ChoiceBox and its field grid.

Also:
- Provides a Checkbox and Radio in ol-components
- Exports styles for reuse (applies to the SearchDisplay course-search-utils overrides).
- Adds Storybook stories for ChoiceBox.
- Removes the hidden input control in ChoiceBox (accessibility concern) to style directly.
- Upgrades Storybook.

### Screenshots (if appropriate):

<img width="765" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/e0358111-5caa-4d75-b514-887a71d30f86">

----

<img width="717" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/55f1aa54-f92e-4349-b722-23c778b43a9e">

----

<img width="1027" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/cfaeb6ad-435c-4c4f-addd-98560891f9d5">

----

<img width="1024" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/428e2828-524b-4403-8cf9-90f205976686">



### How can this be tested?

- While authenticated, navigate to /onboarding and step through the onboarding screens.
- See also the ChoiceBox, ChoiceBoxField and Checkbox stories in Storybook
  - /?path=/story/ol-components-choicebox--checkbox
  - /?path=/story/ol-components-choiceboxfield--checkbox
  - /?path=/story/ol-components-checkbox--simple
